### PR TITLE
Moving all simple_animal AI onto /datum/ai.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
DNM for now, still needs testing.

## Description of changes
- Moves simple_animal wandering and speech/emote vars onto `/datum/ai`.
- Moves simple_animal `do_delayed_life_action()` proc onto `/datum/ai`.
- Repaths faithless to `revenant`, /tg/ drakes to `space_dragon`, and cats to `passive/cat`.
- Folds cat friendship and ghost spotting into passive AI datum.
- Folds `/retaliate` behavior down into `/hostile`.
- Does a lot of general rewriting and adjust mob AI to get it working on the AI datum.

## Why and what will this PR improve
Foundation work for broader AI rewrites. 

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Testing:
- [X] Basic wandering behavior.
- [X] Basic hunting behavior (fox).
- [x] Basic fleeing behavior (deer).
- [x] Basic hostile mob behavior.
- [x] Basic retaliate mob behavior.
- [x] Cat fleeing behavior.
- [x] Cat hunting behavior.
- [x] Cat friendship behavior.
- [x] Giant spider behavior.

**Comment by MistakeNot4892** [comment]
Test log comment, will edit as I test:
- Basic AI behavior (wandering, emoting, speaking) all seem to be functioning as expected.
- Ian dances, goes after food, and wanders. He can't eat food currently due to unrelated edibility bugs.
- Hostile mobs appear to be functioning.
    - They will spot targets, perform their threat emote, then close and attack.
    - Breaking LOS/getting far enough away clears their target.
    - Special effects are applied post-attack.
checking.
- Basic retaliatory mobs are working (clowns).
- Goats get mad, calm down, and eat plants.
- Antlions burrow, lurk and attack when you get close.
- Single shot and rapid fire ranged mobs get into range, then fire.
- Basic giant spiders are working (wandering around, attacking stuff, not attacking each other).
    - Guards and nurses are registering each other and divorcing on death.
    - Nurses dying causes paired guards to go berserk.
    - Nurses will create webbing and cocoon dead mobs.
    - Nurses will lay eggs after cocooning mobs mobs.
    - Hunters will launch themselves at you and knock you down if they have enough distance.
- Deer/rabbits will flee from you.
- Foxes chase and kill/eat rabbits.
- Exoplanet beasts pursue and eat prey when hungry.
- Mega hivebot uses rockets and gun, then switches to melee.
- Commanded mobs receive/process commands.
    - Summoned bears set their master appropriately.
    - Bears will do their dance, follow, and attack.
- King of goats AI seems fine, have not exhaustively tested the other boss mechanics as it's not relevant.
- Faithful hound responds to password and threatens non-friends.
- Parrots are baseline functional.
    - They steal small items.
    - They fly around.
    - They steal stuff and take it back to their perch.

**Comment by MistakeNot4892** [comment]
I -think- that the main body of changes are done now. I've moved almost all mob vars onto the AI datums and written interface procs. Testing/refining shouldn't cause any significant code changes from here.

**Comment by MistakeNot4892** [comment]
Sorry to squash it all into one commit, but there was a lot of back and forth and overlapping changes in there. I don't think this can effectively be broken up since everything kind of ties together.

**Comment by MistakeNot4892** [comment]
All integrated with SSautomove now.

File: code/__defines/ai.dm [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// TODO: FSM/decl based stances
+#define STANCE_NONE                   0
```
Comment (out-of-phaze) [inline code/__defines/ai.dm]: is there a reason these have to be numbers, aside from `STANCE_NONE` being falsey? might be a good idea to, between now and whenever these stances are replaced with decls, change them to strings to reduce collisions esp. with modpacks and such

**Review by out-of-phaze — DISMISSED** [review]

File: code/__defines/ai.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/__defines/ai.dm]: They're just used as state identifiers so no reason that I know of. I could replace them with dummy stance decl types in the leadup to the FSM AI plan?

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**